src/ts/components/subtitleoverlay.ts

```diff
@@ -339,34 +365,52 @@ class ActiveSubtitleManager {
     return id;
   }
 
-  /**
-   * Adds a subtitle cue to the manager and returns the label that should be added to the subtitle overlay.
-   * @param event
-   * @return {SubtitleLabel}
-   */
-  cueEnter(event: SubtitleCueEvent): SubtitleLabel {
-    let id = ActiveSubtitleManager.calculateId(event);
+  cueEnter(event: SubtitleCueEvent, label: SubtitleLabel): void {
+    this.addCueToMap(event, label);
+  }
 
-    let label = new SubtitleLabel({
-      // Prefer the HTML subtitle text if set, else try generating a image tag as string from the image attribute,
-      // else use the plain text
-      text: event.html || ActiveSubtitleManager.generateImageTagText(event.image) || event.text,
-      vtt: event.vtt,
-      region: event.region,
-      regionStyle: event.regionStyle,
-    });
+  cueUpdate(event: SubtitleCueEvent, label: SubtitleLabel): void {
+    this.addCueToMap(event, label);
+  }
+
+  getPreviousCue(event: SubtitleCueEvent): SubtitleLabel {
+    return this.removeCueFromMap(event);
+  }
+
+  private addCueToMap(event: SubtitleCueEvent, label: SubtitleLabel): void {
+    let id = ActiveSubtitleManager.calculateId(event);
 
     // Create array for id if it does not exist
     this.activeSubtitleCueMap[id] = this.activeSubtitleCueMap[id] || [];
 
     // Add cue
     this.activeSubtitleCueMap[id].push({ event, label });
     this.activeSubtitleCueCount++;
+  }
 
-    return label;
+  private removeCueFromMap(event: SubtitleCueEvent): SubtitleLabel {
+    let id = ActiveSubtitleManager.calculateId(event);
+    let activeSubtitleCues = this.activeSubtitleCueMap[id];
+
+    if (activeSubtitleCues && activeSubtitleCues.length > 0) {
+      // Remove cue
+      /* We apply the FIFO approach here and remove the oldest cue from the associated id. When there are multiple cues
+       * with the same id, there is no way to know which one of the cues is to be deleted, so we just hope that FIFO
+       * works fine. Theoretically it can happen that two cues with colliding ids are removed at different times, in
+       * the wrong order. This rare case has yet to be observed. If it ever gets an issue, we can take the unstable
+       * cue end time (which can change between CueEnter and CueExit IN CueUpdate) and use it as an
+       * additional hint to try and remove the correct one of the colliding cues.
+       */
+      let activeSubtitleCue = activeSubtitleCues.shift();
+      this.activeSubtitleCueCount--;
+
+      return activeSubtitleCue.label;
+    } else {
+      return null;
```

Comment:
As per code talk we should prefer undefined over nulls
```suggestion
      return undefined;
```

Reply:
Fixed in 6b9d353